tools/crm_mon.c

```diff
@@ -177,6 +165,289 @@ crm_trigger_t *refresh_trigger = NULL;
 #  define print_as(fmt, args...) fprintf(stdout, fmt, ##args);
 #endif
 
+struct {
+    int reconnect_msec;
+    int fence_history_level;
+    int verbose;
+    gboolean one_shot;
+    gboolean daemonize;
+    gboolean group_by_node;
+    gboolean inactive_resources;
+    gboolean watch_fencing;
+    gboolean show_bans;
+    gboolean print_clone_detail;
+    gboolean print_brief;
+    gboolean print_pending;
+    char *pid_file;
+    char *external_agent;
+    char *external_recipient;
+} options = {
+    .reconnect_msec = 5000,
+    .fence_history_level = 1
+};
+
+static gboolean
+as_cgi_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **error) {
+    if (output_format != mon_output_console) {
+        g_set_error(error, G_OPTION_ERROR, CRM_EX_INVALID_PARAM, "-w requires console output format");
+        return FALSE;
+    }
```

Comment:
FYI, the point of checking output_format == mon_output_console (i.e. the default) is simply to make the format options mutually exclusive (rather than actually requiring console format). I'm not sure it's all that important when we move to the new output model. Does glib have any built-in mechanism for saying certain arguments are exclusive?

Reply:
No, there's no way to specify that.  This would be a behavior change, but we could just say that the last format option given on the command line is the one that takes effect and get rid of all these checks.  Another possibility would be to add mon_output_unset (I think I actually have this in another patch to come) and if output_format is anything other than mon_output_unset in these callbacks, error.  And then after command line processing, we set it to the default if it wasn't set manually.

Reply:
I'm fine with saying the last one wins (which means we don't need the mon_output_console checks in these blocks)

Reply:
One wrinkle is that -w wants -h <filename>, and I feel like we should support it no matter which order the two options are given.